buildSrc/subprojects/ide/src/main/kotlin/org/gradle/gradlebuild/ide/IdePlugin.kt

```diff
@@ -77,230 +56,120 @@ limitations under the License."""
-                    create<Make>("make") {
-                        enabled = false
-                    }
-                }
-            }
             create<Remote>("Remote debug port 5005") {
                 mode = Remote.RemoteMode.ATTACH
                 transport = Remote.RemoteTransport.SOCKET
                 sharedMemoryAddress = "javadebug"
                 host = "localhost"
                 port = 5005
             }
-        }
-    }
-
-    private
-    fun Project.docsProject() =
-        project(":docs")
-
-    private
-    val lang: String
-        get() = System.getenv("LANG") ?: "en_US.UTF-8"
-
-    private
-    fun Project.configureJUnitDefaults() {
-        val rootProject = this
-        val docsProject = docsProject()
-        docsProject.afterEvaluate {
-            rootProject.plugins.withType<IdeaPlugin> {
-                with(model) {
-                    project {
-                        settings {
-                            runConfigurations {
-                                create<JUnit>("defaults") {
-                                    defaults = true
-                                    vmParameters = getDefaultJunitVmParameters(docsProject)
-                                    envs = mapOf("LANG" to lang)
-                                }
-                            }
-                        }
-                    }
-                }
+            create<JUnit>("defaults") {
+                defaults = true
+                envs = mapOf("LANG" to rootProject.lang)
+                lateSetJunitVmParameters(rootProject)
             }
         }
     }
 
-    // TODO this can be removed if we *always* delegate to Gradle
     private
-    fun ProjectSettings.configureSyncTasks(subprojects: Set<Project>) {
-        subprojects.forEach { subproject ->
-            subproject.run {
-                apply(plugin = "idea")
-                afterEvaluate {
-                    taskTriggers {
-                        val classpathManifest = tasks.findByName("classpathManifest")
-                        if (classpathManifest != null) {
-                            afterSync(classpathManifest)
-                        }
-                        when (subproject.name) {
-                            "baseServices" -> afterSync(tasks.getByName("buildReceiptResource"))
-                            "core" -> afterSync(tasks.getByName("pluginsManifest"), tasks.getByName("implementationPluginsManifest"))
-                            "docs" -> afterSync(tasks.getByName("defaultImports"))
-                            "internalIntegTesting" -> afterSync(tasks.getByName("prepareVersionsInfo"))
-                            "kotlinCompilerEmbeddable" -> afterSync(tasks.getByName("classes"))
-                            "kotlinDsl" -> afterSync(tasks["generateExtensions"])
-                        }
-                    }
-                }
+    fun JUnit.lateSetJunitVmParameters(rootProject: Project) {
```

Comment:
We don't need the JUnit vm parameters, since we now use Gradle to run unit tests, right?

Reply:
Right... I thought it is still used but it is not. Well then I'll remove all of this.